spec/queries/keywords_query_spec.rb

```diff
@@ -30,14 +28,78 @@
       end
 
       context 'given filter_params is nil' do
-        it 'returns all keywords' do
-          user = Fabricate(:user)
-          Fabricate.times(2, :keyword, user: user)
+        it 'returns Keyword scope' do
           keywords = described_class.new(Keyword, nil).call
 
-          expect(keywords.count).to eq 2
+          expect(keywords).to eq Keyword
+        end
+      end
+    end
+
+    context 'given the user searchs for result_urls' do
+      context 'given filtering with a word vpn and match at least 1' do
+        it 'returns a keyword' do
+          result_urls = ['https://www.thetopvpn.com', 'https://www.nordvpn.com', 'https://www.vnexpress.net']
+          keyword = Fabricate(:keyword, result_urls: result_urls)
+          filter_params = { word: 'vpn', match_at_least: '1' }
+          keywords = described_class.new(Keyword, filter_params).call
+
+          expect(keywords).to contain_exactly(keyword)
+        end
+      end
+
+      context 'given filtering with a word vpn and match at least 2' do
+        it 'returns a keyword' do
+          result_urls = ['https://www.topvpn.com/vpn', 'https://www.nordvpn.com', 'https://www.vnexpress.net']
+          keyword = Fabricate(:keyword, result_urls: result_urls)
+          filter_params = { word: 'vpn', match_at_least: '2' }
+          keywords = described_class.new(Keyword, filter_params).call
+
+          expect(keywords).to contain_exactly(keyword)
+        end
+      end
+
+      context 'given there is no matched result' do
+        it 'returns empty' do
```

Comment:
```suggestion
        it 'is empty' do
```

or it returns an empty scope